### PR TITLE
Add NextScheduledTime in schedule object

### PR DESCRIPTION
**Pull request by stonezdj**
Add NextSchedule in utils
  Add NextScheduledTime in ScheduleObj to display it in UI

Signed-off-by: stonezdj <stonezdj@gmail.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/common/utils/utils.go [hunk 1]
```diff
@@ -27,6 +27,10 @@ import (
 	"strings"
 	"time"
 
+	cronlib "github.com/robfig/cron"
```
Comment (chlins) [inline src/common/utils/utils.go:30]: Why not use latest v3 version?

File: src/common/utils/utils.go [hunk 2]
```diff
@@ -277,3 +281,20 @@ func FindNamedMatches(regex *regexp.Regexp, str string) map[string]string {
...
+	cr := strings.TrimSpace(cron)
+	if _, err := cronlib.Parse(cron); err != nil {
+		return time.Time{}
+	}
+	parts := strings.Split(cr, " ")
+	// remove the seconds field, it should always be 0 in harbor cron string
+	if len(parts) > 5 {
+		cr = cr[2:]
+	}
+	return cronexpr.MustParse(cr).Next(curTime)
```
Comment (chlins) [inline src/common/utils/utils.go:299]: This function uses two library to parse and compute next time, but I confuse here why not just use `robfig/cron.Parse(cron).Next(curTime)`.

Comment (stonezdj) [reply src/common/utils/utils.go]: Updated

**Review by chlins — APPROVED** [review]
lgtm

File: src/common/utils/utils.go [hunk 3]
```diff
@@ -277,3 +279,16 @@ func FindNamedMatches(regex *regexp.Regexp, str string) map[string]string {
...
+
+// NextSchedule return next scheduled time with a cron string and current time provided
+// the cron string could contain 6 tokens
+// if the cron string is invalid, it returns a zero time
+func NextSchedule(cron string, curTime time.Time) time.Time {
+	cr := strings.TrimSpace(cron)
+	p := cronlib.NewParser(cronlib.Second | cronlib.Minute | cronlib.Hour | cronlib.Dom | cronlib.Month | cronlib.Dow)
+	s, err := p.Parse(cr)
+	if err != nil {
+		return time.Time{}
```
Comment (wy65701436) [inline src/common/utils/utils.go:291]: log error?

File: src/go.mod [hunk 4]
```diff
@@ -50,6 +50,7 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.11.0
 	github.com/robfig/cron v1.0.0
```
Comment (wy65701436) [inline src/go.mod:52]: can we utilize just one of these two?

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/common/utils/utils.go (hunk 3 above)
Comment (stonezdj) [reply src/common/utils/utils.go:290]: Updated

**Pull request merged**